File: Examples/DevCycle-iOS-Example-App-ObjC/DevCycle-iOS-Example-App-ObjC/DevCycleManager.m
//
//  DevCycleManager.m
//  DevCycle-Example-App-ObjC
//
//

#import "DevCycleManager.h"
@import DevCycle;

@interface DevCycleManager()

@property (atomic) DevCycleClient* _Nullable client;

@end

@implementation DevCycleManager

static NSString *const DEVCYCLE_KEY = @"<DEVCYCLE_MOBILE_SDK_KEY>";

+ (id)sharedManager {
    static DevCycleManager *sharedMyManager = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedMyManager = [[self alloc] init];
    });
    return sharedMyManager;
}

- (id)init {
  if (self = [super init]) {
      self.client = nil;
  }
  return self;
}

- (DevCycleClient*)initialize:(DevCycleUser *)user onInitialized:(void (^_Nullable)(NSError*))onInitialized {
    NSError *err = nil;
    
    DevCycleOptions *options = [[DevCycleOptions alloc] init];
//    options.logLevel = LogLevel.debug;
    
    self.client = [DevCycleClient initialize:DEVCYCLE_KEY
                                   user:user
                                options:options
                          onInitialized:onInitialized];
    if (err) {
        NSLog(@"Error Starting DevCycle: %@", err);
    }
    
    return self.client;
}

@end

File: Examples/DevCycle-iOS-Example-App-Swift/DevCycle-iOS-Example-App-Swift/DevCycleManager.swift
//
//  DevCycleManager.swift
//  DevCycle-Example-App
//
//

import DevCycle
import Foundation

struct DevCycleKeys {
    static var DEVELOPMENT = "YOUR_DEVCYCLE_MOBILE_KEY_HERE"
}

class DevCycleManager {

    var client: DevCycleClient?
    static let shared = DevCycleManager()

    func initialize(user: DevCycleUser) {
        let options = DevCycleOptions.builder()
            //                                .logLevel(.debug)
            .build()

        guard
            let client = try? DevCycleClient.builder()
                .sdkKey(DevCycleKeys.DEVELOPMENT)
                .user(user)
                .options(options)
                .build(onInitialized: nil)
        else {
            return
        }
        self.client = client
    }
}
